### PR TITLE
resolved_ts: separate resolevd_ts config from cdc config

### PR DESCRIPTION
**Pull request by NingLin-P**
### What problem does this PR solve?

Problem Summary:

It is weird that the configs used by `resolved-ts` worker are resided in `CdcConfig`.

### What is changed and how it works?

What's Changed:

Use a dedicated config `ResolvedTsConfig` for `resolved-ts` worker, and support dynamically change of the `resolevd-ts.advance-ts-interval` config.

Tests 

- Unit test

### Release note 
- Add resolved-ts config

### CODE REVIEW

File: components/resolved_ts/src/advance.rs [hunk 1]
```diff
@@ -70,32 +59,27 @@ impl<T, E: KvEngine> AdvanceTsWorker<T, E> {
             pd_client,
             worker,
             timer: SteadyTimer::default(),
-            raft_router,
             store_meta,
             concurrency_manager,
-            advance_ts_interval,
-            hibernate_regions_compatible,
```
Comment (BusyJay) [inline components/resolved_ts/src/advance.rs:77]: Why remove this?

**Review by BusyJay — COMMENTED** [review]
The change seems break compatibility.

File: components/resolved_ts/src/advance.rs (hunk 1 above)
Comment (NingLin-P) [reply components/resolved_ts/src/advance.rs:77]: We need `CheckLeader` to sync `safe_ts` from leader to follower, the stale read feature can't work if `hibernate_regions_compatible` is `false`,  so we assume it is `true` if the feature is turned on by `resolved-ts.enable`

File: src/config.rs [hunk 2]
```diff
@@ -2207,16 +2207,13 @@ impl Default for BackupConfig {
     }
 }
 
-// TODO: `CdcConfig` is used by both `cdc` worker and `resolved ts` worker
-// should separate it into two configs
-#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Configuration)]
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct CdcConfig {
```
Comment (BusyJay) [inline src/config.rs:2213]: Why keeps the struct?

**Review by BusyJay — COMMENTED** [review]
Test LGTM as long as compatibility is accepted by cdc.

File: src/config.rs (hunk 2 above)
Comment (NingLin-P) [reply src/config.rs:2213]: There are configs only use in cdc

Comment (BusyJay) [reply src/config.rs:2213]: Then why remove `Configuration` derive?

File: src/config.rs [hunk 3]
```diff
@@ -2207,16 +2207,13 @@ impl Default for BackupConfig {
...
     pub min_ts_interval: ReadableDuration,
```
Comment (BusyJay) [inline src/config.rs:2214]: Is it still a valid configuration given now `advance_ts_interval` is used?

File: src/config.rs (hunk 2 above)
Comment (NingLin-P) [reply src/config.rs:2213]: `Configuration` trait is used for config that can be changed dynamically, `CdcConfig` is not supported to doing so currently.

File: src/config.rs (hunk 3 above)
Comment (NingLin-P) [reply src/config.rs:2214]: `advance_ts_interval` and `min_ts_interval` is used by different config

**Review by Yisaer — APPROVED** [review]

File: components/test_raftstore/src/common-test.toml [hunk 4]
```diff
@@ -110,5 +110,7 @@ num-threads = 1
 wait-for-lock-timeout = "3s"
 wake-up-delay-duration = "100ms"
 
-[cdc]
```
Comment (nolouch) [inline components/test_raftstore/src/common-test.toml:113]: Does this change break compatibility？

**Review by nolouch — COMMENTED** [review]
rest LGTM

File: components/test_raftstore/src/common-test.toml (hunk 4 above)
Comment (NingLin-P) [reply components/test_raftstore/src/common-test.toml:113]: No, this config file is used for test, and the `cdc` config is only used for stale read test

File: components/server/src/server.rs [hunk 5]
```diff
@@ -611,16 +611,29 @@ impl<ER: RaftEngine> TiKVServer<ER> {
...
-        let rts_scheduler = rts_worker.scheduler();
-
         // Register cdc
         let cdc_ob = cdc::CdcObserver::new(cdc_scheduler.clone());
         cdc_ob.register_to(self.coprocessor_host.as_mut().unwrap());
-        // Register resolved ts
-        let resolved_ts_ob = resolved_ts::Observer::new(rts_scheduler.clone());
-        resolved_ts_ob.register_to(self.coprocessor_host.as_mut().unwrap());
+
+        // Create resolved ts worker
```
Comment (5kbpers) [inline components/server/src/server.rs:618]: Could you leave a TODO for adding a cdc config manager here?

**Review by 5kbpers — APPROVED** [review]

**Review by nolouch — COMMENTED** [review]
LGTM. plz update the release note

**Pull request merged**